aspnetcore/blazor/components/prerender.md

```diff
@@ -17,358 +17,134 @@ uid: blazor/components/prerender
 
-To preserve prerendered state, decide what state to persist using the <xref:Microsoft.AspNetCore.Components.PersistentComponentState> service. <xref:Microsoft.AspNetCore.Components.PersistentComponentState.RegisterOnPersisting%2A?displayProperty=nameWithType> registers a callback to persist the component state during prerendering. The state is retrieved when the component renders interactively. Make the call at the end of initialization code in order to avoid a potential race condition during app shutdown.
+* *Recommended* for shared framework services: For shared framework services that merely aren't registered server-side in the main project, register the services in the main project, which makes them available during prerendering. For an example of this scenario, see the guidance for <xref:System.Net.Http.HttpClient> services in <xref:blazor/call-web-api?pivots=webassembly#client-side-services-for-httpclient-fail-during-prerendering>.
 
-The following counter component example persists counter state during prerendering and retrieves the state to initialize the component.
+* *Recommended* for services outside of the shared framework: Create a custom service implementation for the service on the server. Use the service normally in interactive components of the `.Client` project. For a demonstration of this approach, see <xref:blazor/fundamentals/environments#read-the-environment-client-side-in-a-blazor-web-app>.
 
-`PrerenderedCounter2.razor`:
+* Create a service abstraction and create implementations for the service in the `.Client` and server projects. Register the services in each project. Inject the custom service in the component.
 
-:::code language="razor" source="~/../blazor-samples/8.0/BlazorSample_BlazorWebApp/Components/Pages/PrerenderedCounter2.razor":::
-
-When the component executes, `currentCount` is only set once during prerendering. The value is restored when the component is rerendered. The following is example output.
-
-> [!NOTE]
-> If the app adopts [interactive routing](xref:blazor/fundamentals/routing#static-versus-interactive-routing) and the page is reached via an internal [enhanced navigation](xref:blazor/fundamentals/routing#enhanced-navigation-and-form-handling), prerendering doesn't occur. Therefore, you must perform a full page reload for the component to see the following output. For more information, see the [Interactive routing and prerendering](#interactive-routing-and-prerendering) section.
-
-> :::no-loc text="info: BlazorSample.Components.Pages.PrerenderedCounter2[0]":::  
-> :::no-loc text="      currentCount set to 96":::  
-> :::no-loc text="info: BlazorSample.Components.Pages.PrerenderedCounter2[0]":::  
-> :::no-loc text="      currentCount restored to 96":::
+* You might be able to add a `.Client` project package reference to a server-side package and fall back to using the server-side API when prerendering on the server.
```

Comment:
This approach seems risky to recommend. I see a lot of Blazor WebAssembly mistakenly reference server assemblies from their client projects resulting in large client apps. I think we should remove this one.

```suggestion
```